services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crossregion/endpointprovider/BucketEndpointProvider.java

```diff
@@ -43,9 +43,16 @@ public static BucketEndpointProvider create(S3EndpointProvider delegateEndPointP
     @Override
     public CompletableFuture<Endpoint> resolveEndpoint(S3EndpointParams endpointParams) {
         Region crossRegion = regionSupplier.get();
-        return delegateEndPointProvider.resolveEndpoint(
-            endpointParams.copy(c -> c.region(crossRegion == null ? endpointParams.region() : crossRegion)
-                                      .useGlobalEndpoint(false)));
+        S3EndpointParams.Builder endpointParamsBuilder = endpointParams.toBuilder();
+        if (crossRegion != null) {
+            endpointParamsBuilder.region(crossRegion);
+        } else {
+            if (Region.AWS_GLOBAL.equals(endpointParams.region())) {
+                endpointParamsBuilder.region(Region.US_EAST_1);
+            }
+            endpointParamsBuilder.useGlobalEndpoint(false);
```

Comment:
Can we add a comments on why these parts are needed? I'm not sure why we were setting use-global-endpoint to false in the first place, and now I really don't get what all this is for without reading the history.